aptos-move/framework/aptos-framework/sources/configs/gas_schedule.move

```diff
@@ -23,16 +31,27 @@ module aptos_framework::gas_schedule {
         entries: vector<GasEntry>
     }
 
+    /// Called during a module upgrade.
+    /// Creates and publishes the resources that are added at a later time.
+    public fun upgrade_module(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
```

Comment:
The idea I suggested is to let this be callable by anyone. I.e. remove this check. There should be no harm to let anyone call this as:
1. It can only be called once. Calling it more wouldn't do anything as GasFeatureVersion has already been initialized.
2. Timing doesn't really matter here as GasFeatureVersion is not used anywhere except for set_gas_feature_version and may be some blockchain-side code, which would be gated by on chain flags anyway. So in other words, further usage would need to go through governance voting anyway. Therefore when this function is called exactly wouldn't matter. We can make sure this is called as soon as possible after the upgrade proposal goes in.

Reply:
> There should be no harm to let anyone call this as:
But we need a signer representing the aptos_framework anyways. 

> some blockchain-side code, which would be gated by on chain flags
This resource will be almost exclusively used by the VM. So you are saying that I should gate this by a feature flag? I'm not so sure since this gas feature version is a system parallel to the feature flags...

Reply:
I missed this earlier. Unfortunately since this still requires the 0x1 signer, this would need to be done in a separate governance proposal after the upgrade one.

Reply:
Is this supposed to be called to initialize this module here when it is not added via genesis? This would make the upgrade process even more complex.  Alternatives:

- `init_module`
-  on the fly in `set_gas_schedule`, and return a default in `get_gas_version` if resource does not exist.
-  not needed because piggy packed with gas schedule.